charts/kube-prometheus-stack/values.yaml

```diff
@@ -3683,22 +3683,24 @@ prometheus:
     #   matchLabels:
     #     prometheus: somelabel
 
-    ## If true, a nil or {} value for prometheus.prometheusSpec.scrapeConfigSelector will cause the
-    ## prometheus resource to be created with selectors based on values in the helm deployment,
-    ## which will also match the scrapeConfigs created
-    ##
-    scrapeConfigSelectorNilUsesHelmValues: true
-
     ## scrapeConfigs to be selected for target discovery.
-    ## If {}, select all scrapeConfigs
-    ##
-    scrapeConfigSelector: {}
+    ## If matchLabels.release: "{{ $.Release.Name }}" the prometheus resource will be created
+    ## with selectors based on values in the helm deployment, which will also match the scrapeConfigs created
+    ## To remove matchLabels from the selector condition, explicitly set matchLabels to null.
+    ## If no other selectors are configured, prometheus-operator will select all scrapeConfigs.
+    ## To remove the release label from the matchLabels condition, explicit set release to null. 
+    ## If null, exclude the field from the prometheusSpec (keeping  downward compatibility with older versions of CRD)
+    ##
+    scrapeConfigSelector:
+      matchLabels:
+        release: "{{ $.Release.Name }}"
     ## Example which selects scrapeConfigs with label "prometheus" set to "somelabel"
     # scrapeConfigSelector:
     #   matchLabels:
     #     prometheus: somelabel
 
     ## If nil, select own namespace. Namespaces to be selected for scrapeConfig discovery.
```

Comment:
Shouldn't this be `If empty, select own namespace`?
`scrapeConfigNamespaceSelector: {}` empty object means select own NS?

Reply:
https://github.com/prometheus-operator/prometheus-operator/blob/781d6df6bc8e48b010a74348f19526f66c5e2c54/pkg/apis/monitoring/v1/prometheus_types.go#L177-L179


```
	// Namespaces to match for ScrapeConfig discovery. An empty label selector
	// matches all namespaces. A null label selector matches the current
	// namespace only.
```

Thats what I understand here.